### ISSUE TITLE
More flexibility on Spice reader

### ISSUE THREAD

**Issue by klayoutmatthias** [KLayout/klayout#798]
This follows the discussion here: https://www.klayout.de/forum/discussion/1812/lvs-netlistspicereader-3-terminal-resistor-confusion#latest

Specifically two requests arise from this discussion:
* support R models with R=value specification
* SpiceReaderDelegate enhanced to allow translation of generic lines not following the internal syntax imposed by the element

**Comment by klayoutmatthias** [comment]
I have added some more flexibility to the Spice reader. Apart from a few new methods for the delegate and the ability to handle more than two nets for R, C and (not used yet) L, the following 2- and 3-terminal resistor flavors are supported:

2-terminal: 

```
R$2 3 4 1.7k
R$4 3 4 1.7k RMODEL
R$5 3 4 RMODEL R=1.7k
```

3-terminal: 

```
R$6 3 4 5 RMODEL2 R=1.7k
R$7 3 4 5 1.7k RMODEL2
R$8 3 4 5 1.7k 
```

Some for caps.

**Comment by klayoutmatthias** [comment]
New NetlistSpiceReaderDelegate methods are:

* "control_statement": receives a control statement (virtual)
* "parse_element": low-level parsing of an element line (virtual)
* "parse_element_components": a helper method for splitting Spice lines
* "translate_net_name": for normalizing net names (virtual)
* "value_from_string": a helper method for turning strings (also formulas) into numeric values

For details see here: https://www.klayout.de/doc-qt5/code/class_NetlistSpiceReaderDelegate.html#k_1

These features have been released in version 0.27.1.
